src/main/java/seedu/address/storage/JsonAdaptedOrder.java

```diff
@@ -44,7 +53,7 @@ public JsonAdaptedOrder(@JsonProperty("orderNumber") String orderNumber,
     public JsonAdaptedOrder(Order order) {
         this.orderNumber = order.getOrderNumber().value;
         this.person = new JsonAdaptedPerson(order.getPerson());
-        this.medicineName = order.getMedicineName();
+        medicines.addAll(order.getMedicines().stream().map(Medicine::getMedicineName).collect(Collectors.toList()));
```

Comment:
From JsonAdaptedPerson:        
```
tags.addAll(source.getTags().stream()
                .map(JsonAdaptedTag::new)
                .collect(Collectors.toList()));
```

Can employ similar method to get list of Medicine objects